benchmarks/asv_delegated.py

```diff
@@ -0,0 +1,350 @@
+# Copyright Iris contributors
+#
+# This file is part of Iris and is released under the BSD license.
+# See LICENSE in the root of the repository for full licensing details.
+"""ASV plug-in providing an alternative :class:`asv.environments.Environment` subclass.
+
+Preps an environment via custom user scripts, then uses that as the
+benchmarking environment.
+
+"""
+
+from contextlib import contextmanager, suppress
+from os import environ
+from os.path import getmtime
+from pathlib import Path
+import sys
+
+from asv import util as asv_util
+from asv.console import log
+from asv.environment import Environment, EnvironmentUnavailable
+from asv.repo import Repo
+from asv.util import ProcessError
+
+
+class EnvPrepCommands:
+    """A container for the environment preparation commands for a given commit.
+
+    Designed to read a value from the `delegated_env_commands` in the ASV
+    config, and validate that the command(s) are structured correctly.
+    """
+
+    ENV_PARENT_VAR = "ENV_PARENT"
+    env_parent: Path
+    commands: list[str]
+
+    def __init__(self, environment: Environment, raw_commands: tuple[str]):
+        env_var = self.ENV_PARENT_VAR
+        raw_commands = list(raw_commands)
+
+        (first_command,) = environment._interpolate_commands(raw_commands[0])
+        command, env, return_codes, cwd = first_command
```

Comment:
Following [the pattern in ASV's own source](https://github.com/airspeed-velocity/asv/blob/8877eff343807e8bc594690dec08bc0da2caa401/asv/environment.py#L821).